hyperliquid/utils/types.py

```diff
@@ -29,6 +29,14 @@
 )
 SpotMeta = TypedDict("SpotMeta", {"universe": List[SpotAssetInfo], "tokens": List[SpotTokenInfo]})
 
+
+SpotdAssetCtxsTokenInfo = TypedDict("SpotdAssetCtxsTokenInfo", {"name": str, "szDecimals": int, "weiDecimals": int})
+SpotdAssetCtxsUniverseInfo = TypedDict("SpotdAssetCtxsUniverseInfo", {"name": str, "tokens": List[int]})
+SpotdAssetCtxsMarketDataInfo = TypedDict("SpotdAssetCtxsMarketDataInfo", {"dayNtlVlm": str, "markPx": str, "midPx": str, "prevDayPx": str})
```

Comment:
name this SpotAssetCtx. Why did you add these lowercase d's?
```suggestion
SpotAssetCtx = TypedDict("SpotAssetCtx", {"dayNtlVlm": str, "markPx": str, "midPx": Optional[str], "prevDayPx": str, "circulatingSupply": str, "coin": str})
```

Reply:
idk where you got the "circulatingSupply": str, "coin": str part

Reply:
You can look at the returned data

Reply:
I implemented some of the changes